File: app/src/main/java/loopeer/com/appbarlayout_spring_extension/ItemFragment.java
package loopeer.com.appbarlayout_spring_extension;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import loopeer.com.appbarlayout_spring_extension.dummy.DummyContent;

public class ItemFragment extends Fragment {

    public static ItemFragment newInstance() {
        ItemFragment fragment = new ItemFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_item_list, container, false);
        RecyclerView recyclerView = (RecyclerView) view;
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setAdapter(new MyItemRecyclerViewAdapter(DummyContent.ITEMS));
        return view;
    }


}

File: app/src/main/java/loopeer/com/appbarlayout_spring_extension/ItemFragment.kt
package loopeer.com.appbarlayout_spring_extension

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import loopeer.com.appbarlayout_spring_extension.dummy.DummyContent

class ItemFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_item_list, container, false)
        val recyclerView = view as RecyclerView
        recyclerView.layoutManager = LinearLayoutManager(context)
        recyclerView.adapter = MyItemRecyclerViewAdapter(DummyContent.ITEMS)
        return view
    }

    companion object {

        fun newInstance(): ItemFragment {
            val fragment = ItemFragment()
            val args = Bundle()
            fragment.arguments = args
            return fragment
        }
    }


}
